packages/node_modules/pouchdb-core/src/adapter.js

```diff
@@ -508,6 +509,51 @@ AbstractPouchDB.prototype._compact = function (opts, callback) {
     .on('error', callback);
 };
 
+AbstractPouchDB.prototype.purge = adapterFun('_purge', function (docId, rev) {
+  if (typeof this._purge === 'undefined') {
+    throw new Error('Purge is not implemented by the ' + this.adapter + ' adapter.');
+  }
+  var self = this;
+
+  return new Promise((resolve) => {
+    self._getRevisionTree(docId, async (whatever, revs) => {
+      if (!revs) {
+        throw new Error('The target document doesn’t exist (has no revisions)');
+      }
+      // `findPathToLeaf` throws if rev is invalid
+      const path = findPathToLeaf(revs, rev);
+      console.log("purge path", path);
+      await (async () =>
```

Comment:
More out of curiosity: What is the reason for wrapping the `_purge` call into this `await (async () => {...})()` call? Is this done in order to push the execution on another thread or what is the background of this?

Reply:
A misunderstanding on my part, this has all been changed to use `callback()` correctly, and the awaits and promises removed.